File transfer fails on large files

Xubuntu 18.04 64-bit
firejail self build from https://github.com/netblue30/firejail/commit/9eca281abbb24f97fdb92c06018cd2bd0a5b215e
```
$ firejail --version
firejail version 0.9.57

Compile time support:
        - AppArmor support is enabled
        - AppImage support is enabled
        - chroot support is enabled
        - file and directory whitelisting support is enabled
        - file transfer support is enabled
        - networking support is enabled
        - overlayfs support is enabled
        - private-home support is enabled
        - seccomp-bpf support is enabled
        - user namespace support is enabled
        - X11 sandboxing support is enabled


```

**Steps to reproduce**

1) check free space available in `/run` with `df -Th`, and create or find a file larger than that (On my system I used a 2 GB file)

2) run `firejail --noprofile --overlay-tmpfs`

3) Try to get the large file from (1) using `firejail --get=<pid> path/to/the/file`

**Expected results**

At (3), file should be copied from the overlay-tmpfs to the working directory

**Actual results**

At (3) it sits for a short time, then exits with status 1 without anything seeming to happen.  I think it's related to the file being bigger than free space in `/run` because of this -

https://github.com/netblue30/firejail/blob/9eca281abbb24f97fdb92c06018cd2bd0a5b215e/src/firejail/ls.c#L316-L345

And indeed, a file does appear at `/run/firejail/tmpget-...` while the `firejail --get=...` command is running.

I'm unable to check it because build from 6ac40a597939989a55ab5465f548489e1ea4937d fails:
```
strip: 'src/man/preproc': No such file
make[1]: *** [Makefile:157: install-strip] Error 1
make: *** [Makefile:196: deb] Error 2
```

--- a/src/firejail/ls.c
+++ b/src/firejail/ls.c
@@ -26,6 +26,7 @@
 #include <dirent.h>
 #include <pwd.h>
 #include <grp.h>
+#include <fcntl.h>
 //#include <dirent.h>
 //#include <stdio.h>
 //#include <stdlib.h>
@@ -291,6 +292,41 @@ void sandboxfs(int op, pid_t pid, const char *path1, const char *path2) {
 		printf("file2 %s\n", fname2 ? fname2 : "(null)");
 	}
 
+	// get file from sandbox and store it in the current directory
+	// implemented using --cat
+	if (op == SANDBOX_FS_GET) {
+		char *dest_fname = strrchr(fname1, '/');
+		if (!dest_fname || *(++dest_fname) == '\0') {
+			fprintf(stderr, "Error: invalid file name %s\n", fname1);
+			exit(1);
+		}
+		// create destination file if necessary
+		EUID_ASSERT();
+		int fd = open(dest_fname, O_WRONLY|O_CREAT|O_CLOEXEC, S_IRUSR | S_IWRITE);
+		if (fd == -1) {
+			fprintf(stderr, "Error: cannot open %s for writing\n", dest_fname);
+			exit(1);
+		}
+		struct stat s;
+		if (fstat(fd, &s) == -1)
+			errExit("fstat");
+		if (!S_ISREG(s.st_mode)) {
+			fprintf(stderr, "Error: %s is no regular file\n", dest_fname);
+			exit(1);
+		}
+		if (ftruncate(fd, 0) == -1)
+			errExit("ftruncate");
+		// go quiet - messages on stdout will corrupt the file
+		arg_debug = 0;
+		arg_quiet = 1;
+		// redirection
+		if (dup2(fd, STDOUT_FILENO) == -1)
+			errExit("dup2");
+		assert(fd != STDOUT_FILENO);
+		close(fd);
+		op = SANDBOX_FS_CAT;
+	}
+
 	// sandbox root directory
 	char *rootdir;
 	if (asprintf(&rootdir, "/proc/%d/root", pid) == -1)
@@ -326,92 +362,6 @@ void sandboxfs(int op, pid_t pid, const char *path1, const char *path2) {
 		if (WIFEXITED(status) && WEXITSTATUS(status) == 0);
 		else
 			exit(1);
-	}
-
-	// get file from sandbox and store it in the current directory
-	else if (op == SANDBOX_FS_GET) {
-		char *src_fname =fname1;
-		char *dest_fname = strrchr(fname1, '/');
-		if (!dest_fname || *(++dest_fname) == '\0') {
-			fprintf(stderr, "Error: invalid file name %s\n", fname1);
-			exit(1);
-		}
-
-		EUID_ROOT();
-		if (arg_debug)
-			printf("copy %s to %s\n", src_fname, dest_fname);
-
-		// create a user-owned temporary file in /run/firejail directory
-		char tmp_fname[] = "/run/firejail/tmpget-XXXXXX";
-		int fd = mkstemp(tmp_fname);
-		if (fd == -1) {
-			fprintf(stderr, "Error: cannot create temporary file %s\n", tmp_fname);
-			exit(1);
-		}
-		SET_PERMS_FD(fd, getuid(), getgid(), 0600);
-		close(fd);
-
-		// copy the source file into the temporary file - we need to chroot
-		pid_t child = fork();
-		if (child < 0)
-			errExit("fork");
-		if (child == 0) {
-			// chroot
-			if (chroot(rootdir) < 0)
-				errExit("chroot");
-			if (chdir("/") < 0)
-				errExit("chdir");
-
-			// drop privileges
-			drop_privs(0);
-
-			// copy the file
-			if (copy_file(src_fname, tmp_fname, getuid(), getgid(), 0600)) // already a regular user
-				_exit(1);
-#ifdef HAVE_GCOV
-			__gcov_flush();
-#endif
-			_exit(0);
-		}
-
-		// wait for the child to finish
-		int status = 0;
-		waitpid(child, &status, 0);
-		if (WIFEXITED(status) && WEXITSTATUS(status) == 0);
-		else {
-			unlink(tmp_fname);
-			exit(1);
-		}
-
-		// copy the temporary file into the destination file
-		child = fork();
-		if (child < 0)
-			errExit("fork");
-		if (child == 0) {
-			// drop privileges
-			drop_privs(0);
-
-			// copy the file
-			if (copy_file(tmp_fname, dest_fname, getuid(), getgid(), 0600)) // already a regular user
-				_exit(1);
-#ifdef HAVE_GCOV
-			__gcov_flush();
-#endif
-			_exit(0);
-		}
-
-		// wait for the child to finish
-		status = 0;
-		waitpid(child, &status, 0);
-		if (WIFEXITED(status) && WEXITSTATUS(status) == 0);
-		else {
-			unlink(tmp_fname);
-			exit(1);
-		}
-
-		// remove the temporary file
-		unlink(tmp_fname);
-		EUID_USER();
 	}
 	// get file from host and store it in the sandbox
 	else if (op == SANDBOX_FS_PUT && path2) {

